app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+
+    protected $os = null;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Creates a new Laravel application';
+
+    /**
+     * Create a new command instance.
+     */
+    public function __construct(ExecutableFinder $finder)
+    {
+        parent::__construct();
+
+        $this->cwd = getcwd();
+        $this->basepath = $this->cwd;
+
+        $this->os = new Detector;
+        $this->finder = $finder;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle(): void
+    {
+        $this->setup();
+
+        if (! $this->hasTool('laravel')) {
+            $this->error('Unable to find laravel installer so I must exit. One day I will use composer here instead of exiting.');
+            exit;
+        }
+
+        if (is_dir($this->projectpath)) {
+            if (! $this->askToAndRemoveProject()) {
+                $this->error('Goodbye!');
+                exit;
+            }
+        }
+
+        if ($this->option('dev')) {
+            $this->warn('The laravel installation will use the latest developmental branch by passing in --dev');
+            $branch = ' --dev';
+        } else {
+            $branch = '';
+        }
+
+        $command = "laravel new {$this->projectname}$branch";
+
+        $this->info("Creating a new project named {$this->projectname}");
+        $this->info("Executing command '$command' in directory {$this->basepath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->basepath);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        // @todo Determine why the above outputs before this point; so the following getOutput() call does nothing
+        $this->line($process->getOutput());
+
+        if ($this->replaceEnvVariables()) {
+            $this->info('I replaced .env variables in your new Laravel application');
+        }
+
+        if ($this->option('auth')) {
+            $this->doAuth();
+        }
+
+        if ($this->option('createdb')) {
+            $this->chooseDatabase($this->option('createdb'));
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->hasTool('valet')) {
+            if ($this->option('link')) {
+                $this->doValetLink();
+            }
+            $this->openBrowser();
+        }
+
+        if ($this->hasTool('git')) {
+            $this->doGit();
+        }
+
+        $this->openTextEditor();
+
+        $this->info("You're ready to go! Remember to cd into '{$this->projectpath}' before you start editing.");
+    }
+
+
+    protected function setup()
+    {
+        $this->projectname = $this->argument('name');
+        $this->projecturl = 'http://' . $this->projectname . '.dev'; // @todo .dev is no longer good
```

Comment:
I believe the command line version gets this from your Valet configuration, so we should get that in here as well (plenty of people use custom TLDs in Valet)

Reply:
Thanks, fixed!